psi4/driver/procrouting/scf_proc/scf_iterator.py

```diff
@@ -0,0 +1,766 @@
+            core.print_out("    or increasing MAX_ATTEMPTS (not available for PK integrals).\n")
+
+    # At this point, we are not doing any more SCF cycles
+    #   and we can compute and print final quantities.
+
+    if hasattr(self.molecule(), 'EFP'):
+        efpobj = self.molecule().EFP
+
+        efpobj.compute()  # do_gradient=do_gradient)
+        efpene = efpobj.get_energy(label='psi')
+        efp_wfn_independent_energy = efpene['total'] - efpene['ind']
+        self.set_energies("EFP", efpene['total'])
+
+        SCFE = self.get_energies("Total Energy")
+        SCFE += efp_wfn_independent_energy
+        self.set_energies("Total Energy", SCFE)
+        core.print_out(efpobj.energy_summary(scfefp=SCFE, label='psi'))
+
+        core.set_variable('EFP ELST ENERGY', efpene['electrostatic'] + efpene['charge_penetration'] + efpene['electrostatic_point_charges'])
+        core.set_variable('EFP IND ENERGY', efpene['polarization'])
+        core.set_variable('EFP DISP ENERGY', efpene['dispersion'])
+        core.set_variable('EFP EXCH ENERGY', efpene['exchange_repulsion'])
+        core.set_variable('EFP TOTAL ENERGY', efpene['total'])
+        core.set_variable('CURRENT ENERGY', efpene['total'])
+
+    core.print_out("\n  ==> Post-Iterations <==\n\n")
+
+    self.check_phases()
+    self.compute_spin_contamination()
+    self.frac_renormalize()
+    reference = core.get_option("SCF", "REFERENCE")
+
+    energy = self.get_energies("Total Energy")
+
+#    fail_on_maxiter = core.get_option("SCF", "FAIL_ON_MAXITER")
+#    if converged or not fail_on_maxiter:
+#
+#        if print_lvl > 0:
+#            self.print_orbitals()
+#
+#        if converged:
+#            core.print_out("  Energy converged.\n\n")
+#        else:
+#            core.print_out("  Energy did not converge, but proceeding anyway.\n\n")
+
+    if core.get_option('SCF', 'PRINT') > 0:
+        self.print_orbitals()
+
+    is_dfjk = core.get_global_option('SCF_TYPE').endswith('DF')
+    core.print_out("  @%s%s Final Energy: %20.14f" % ('DF-' if is_dfjk else '', reference, energy))
+    # if (perturb_h_) {
+    #     core.print_out(" with %f %f %f perturbation" %
+    #                    (dipole_field_strength_[0], dipole_field_strength_[1], dipole_field_strength_[2]))
+    # }
+    core.print_out("\n\n")
+    self.print_energies()
+
+    self.clear_external_potentials()
+    if core.get_option('SCF', 'PCM'):
+        calc_type = core.PCM.CalcType.Total
+        if core.get_option("PCM", "PCM_SCF_TYPE") == "SEPARATE":
+            calc_type = core.PCM.CalcType.NucAndEle
+        Dt = self.Da().clone()
+        Dt.add(self.Db())
+        _, Vpcm = self.get_PCM().compute_PCM_terms(Dt, calc_type)
+        self.push_back_external_potential(Vpcm)
+
+    # recompute the Fock matrices as they are modified during the SCF
+    #   iteration and might need to be dumped to checkpoint later
+    self.form_F()
```

Comment:
Is this normal? Seems like a needed JK build. The Fock matrix should be fine I think?

Reply:
@robertodr, this `form_F` used to [_precede_ the pcmsolver part] (https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libscf_solver/hf.cc#L592-L615). now it follows pcmsolver. All the pcm tests run fine with it commented out. Do you recall if it's related to you and if you still need it?

Reply:
If tests work then I guess it was not needed :) I can't recall if I did that or not, but the way I modified `form_F` should always included the PCM part correctly.